### PR TITLE
Button to add new Entity List

### PR DESCRIPTION
**Pull request by ktuite**
Closes https://github.com/getodk/central/issues/575

Screenshot 2024-04-02 at 11 21 17 AM
Screenshot 2024-04-02 at 11 21 20 AM

#### Before submitting this PR, please make sure you have:

- [ ] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: src/components/dataset/list.vue [hunk 1]
```diff
@@ -63,7 +89,10 @@ datasets.request({
         // A brief introduction to Entities shown above Entity Lists for the current Project
         "Entities let you share information between Forms so you can collect longitudinal data, manage cases over time, and represent other workflows with multiple steps.",
         "Entities are created through form design and can be attached to any Form."
-      ]
+      ],
+      // This is the text of a button that is used to create a new Entity List
```
Comment (ktuite) [inline src/components/dataset/list.vue:93]: Period at end of comment.

File: src/components/dataset/new.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        <div class="modal-introduction">
+          <p>{{ $t('introduction[0]') }}</p>
+          <i18n-t tag="p" keypath="moreInfo.helpArticle.full">
+            <template #helpArticle>
+              <doc-link to="central-entities/">{{ $t('moreInfo.helpArticle.helpArticle') }}</doc-link>
+            </template>
+          </i18n-t>
+        </div>
+        <form @submit.prevent="submit">
+          <form-group ref="name.value" v-model.trim="name"
```
Comment (ktuite) [inline src/components/dataset/new.vue:27]: Use two refs, one for the FormGroup component and one for the name data.
```
ref="nameGroup"
v-model.trim="name"
...
const nameGroup = ref(null);
const name = ref('');
```

...

```
@shown = 'nameGroup.focus()'
```

File: src/components/dataset/new.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+}
+</style>
+
+<i18n lang="json5">
+{
+  "en": {
+    // This is the title at the top of a pop-up.
+    "title": "Create Entity List",
+    "introduction": [
+      "Entities let you share information between Forms so you can collect longitudinal data, manage cases over time, and represent other workflows with multiple steps."
```
Comment (ktuite) [inline src/components/dataset/new.vue:142]: Ask issa or Aly what this text should say.
If keeping this text, use the transifex key to reuse existing translation.

File: src/components/dataset/list.vue [hunk 4]
```diff
@@ -11,6 +11,16 @@ except according to the terms contained in the LICENSE file.
 -->
 <template>
   <div id="dataset-list">
+    <page-section>
+      <template #heading>
+        <span>{{ $t('resource.entityLists') }}</span>
+        <button v-if="currentUser.can('dataset.create')"
```
Comment (ktuite) [inline src/components/dataset/list.vue:17]: Should be `projectPermits` and `project != null`

File: test/components/dataset/new.spec.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        })
+        .respondWithData(() => testData.extendedDatasets.createPast(1, { name: newDataset }).last());
+
+    it('shows new name on second screen of modal', async () => {
+      const app = await submit('MyNewDataset');
+      const p = app.findComponent(DatasetNew).find('#entity-list-new-success');
+      p.text().should.containEql('MyNewDataset');
+    });
+
+    // TODO: it should redirect to page for new entity list
```
Comment (ktuite) [inline test/components/dataset/new.spec.js:79]: Look at route path `app.vm.$route.path`

**Review by ktuite — COMMENTED** [review]
Interactive review with @matthew-white

**Review by matthew-white — APPROVED** [review]

**Pull request merged**